File: ruby/lib/steam/community/alien_swarm/alien_swarm_weapon.rb
# This code is free software; you can redistribute it and/or modify it under
# the terms of the new BSD License.
#
# Copyright (c) 2010-2011, Sebastian Staudt

require 'steam/community/game_weapon'

# This class holds statistical information about weapons used by a player
# in Alien Swarm
#
# @author Sebastian Staudt
class AlienSwarmWeapon

  include GameWeapon

  # Returns the accuracy of the player with this weapon
  #
  # @return [Float] The accuracy of the player with this weapon
  attr_reader :accuracy

  # Returns the damage achieved with this weapon
  #
  # @return [Fixnum] The damage achieved with this weapon
  attr_reader :damage

  # Returns the damage dealt to team mates with this weapon
  #
  # @return [Fixnum] The damage dealt to team mates with this weapon
  attr_reader :friendly_fire

  # Returns the name of this weapon
  #
  # @return [String] The name of this weapon
  attr_reader :name

  # Creates a new weapon instance based on the assigned weapon XML data
  #
  # @param [REXML::Element] weapon_data The data representing this weapon
  def initialize(weapon_data)
    super

    @accuracy      = weapon_data.elements['accuracy'].text.to_f
    @damage        = weapon_data.elements['damage'].text.to_i
    @friendly_fire = weapon_data.elements['friendlyfire'].text.to_i
    @name          = weapon_data.elements['name'].text
    @shots         = weapon_data.elements['shotsfired'].text.to_i
  end

end

File: php/lib/steam/community/alien_swarm/AlienSwarmWeapon.php
<?php
/**
 * This code is free software; you can redistribute it and/or modify it under
 * the terms of the new BSD License.
 *
 * Copyright (c) 2010, Sebastian Staudt
 *
 * @author     Sebastian Staudt
 * @license    http://www.opensource.org/licenses/bsd-license.php New BSD License
 * @package    Steam Condenser (PHP)
 * @subpackage Steam Community
 */

require_once STEAM_CONDENSER_PATH . 'steam/community/GameWeapon.php';

/**
 * AlienSwarmWeapon holds statistical information about weapons used by a
 * player in Alien Swarm.
 */
class AlienSwarmWeapon extends GameWeapon {

    private $accuracy;

    private $damage;

    private $friendlyFire;

    private $name;

    /**
     * Creates a new instance of AlienSwarmWeapon based on the assigned weapon
     * XML data
     *
     * @param  $weaponData
     */
    public function __construct($weaponData) {
        parent::__construct($weaponData);

        $this->accuracy     = (float) $weaponData->accuracy;
        $this->damage       = (int) $weaponData->damage;
        $this->friendlyFire = (int) $weaponData->friendlyfire;
        $this->name         = (string) $weaponData->name;
        $this->shots        = (int) $weaponData->shotsfired;
    }

    public function getAccuracy() {
        return $this->accuracy;
    }

    public function getDamage() {
        return $this->damage;
    }

    public function getFriendlyFire() {
        return $this->friendlyFire;
    }

    public function getName() {
        return $this->name;
    }

}
